hooks/usePoll.ts

```
@@ -10,47 +16,60 @@ interface UsePollParams {
-    return useQuery({
-      queryKey: ["poll", id],
-      queryFn: async () => {
-        const res = await fetch(`/poll/${id}`);
-        if (!res.ok) throw new Error("Failed to fetch poll details");
-        return res.json();
-      },
-      enabled: !!id,
-    });
-  };
+export const getPollDetails = (id: number): UseQueryResult<IPollDetails> => {
+  return useQuery({
```

ESLint

[error] React Hook "useQuery" is called in function "getPollDetails" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".

(react-hooks/rules-of-hooks)